src/torrent.cpp

```diff
@@ -1138,6 +1138,20 @@ bool is_downloading_state(int const st)
 		if (m_peer_list) m_peer_list->clear();
 	}
 
+	void torrent::set_sequential_range(piece_index_t start_piece, piece_index_t end_piece)
```

Comment:
a bit of a nit, but assuming this range is defined as `[start_piece, end_piece)`, I think using the same terminology as the standard library would make sense. `begin` and `end`.

It there's a really good reason to define this as a closed range, I think `first` and `last` are good names.